File: ios/Tehilim/Core/Models/LifeCase.swift
import Foundation

struct LifeCase: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let symbol: String
    let note: String
    let psalms: [Int]
    /// Section logique pour l'affichage (optionnelle, fallback "Autres").
    let section: String?
    /// Traductions anglaises (optionnelles, fallback sur la version française).
    let titleEN: String?
    let noteEN: String?
    let sectionEN: String?
    /// Traductions hébraïques (optionnelles, fallback EN puis FR).
    let titleHE: String?
    let noteHE: String?
    let sectionHE: String?
}

extension LifeCase {
    /// Renvoie le titre dans la langue active de l'app.
    var localizedTitle: String {
        LifeCase.pick(fr: title, en: titleEN, he: titleHE)
    }

    /// Renvoie la note dans la langue active de l'app.
    var localizedNote: String {
        LifeCase.pick(fr: note, en: noteEN, he: noteHE)
    }

    /// Section dans la langue active de l'app.
    var localizedSection: String? {
        switch LifeCase.contentCode {
        case "he": return sectionHE ?? sectionEN ?? section
        case "en": return sectionEN ?? section
        default:   return section
        }
    }

    /// Code de langue du contenu éditorial : "fr" | "en" | "he".
    ///
    /// V1.12 — délègue à `AppLocale` (source unique). Avant, le cas `.system`
    /// lisait `Locale.current` (instantané figé) → le contenu restait en
    /// anglais après une bascule à chaud vers Système.
    /// V2.2.b — l'hébreu a désormais son propre contenu (titleHE/noteHE…),
    /// avec repli sur EN puis FR si une clé manque.
    static var contentCode: String { AppLocale.code }

    /// Choisit la chaîne selon la langue active, avec repli HE → EN → FR.
    static func pick(fr: String, en: String?, he: String?) -> String {
        switch contentCode {
        case "he": return he ?? en ?? fr
        case "en": return en ?? fr
        default:   return fr
        }
    }
}

File: android/app/src/main/java/com/david/tehilim/core/model/LifeCase.kt
package com.david.tehilim.core.model

import kotlinx.serialization.Serializable

/**
 * Un « Cas de la vie » — catégorie thématique de lecture des Tehilim.
 * Ex. « Santé », « Voyage », « Parnassa »... Chaque cas a une liste de Tehilim
 * traditionnellement recommandés + un conseil.
 */
@Serializable
data class LifeCase(
    val id: String,
    val title: String,
    val titleEN: String? = null,
    val titleHE: String? = null,
    val symbol: String,
    val note: String,
    val noteEN: String? = null,
    val noteHE: String? = null,
    val section: String? = null,
    val sectionEN: String? = null,
    val sectionHE: String? = null,
    val psalms: List<Int>
) {
    /** Localise avec repli HE → EN → FR. */
    fun localizedTitle(language: ContentLanguage): String = when (language) {
        ContentLanguage.HE -> titleHE ?: titleEN ?: title
        ContentLanguage.EN -> titleEN ?: title
        ContentLanguage.FR -> title
    }

    fun localizedNote(language: ContentLanguage): String = when (language) {
        ContentLanguage.HE -> noteHE ?: noteEN ?: note
        ContentLanguage.EN -> noteEN ?: note
        ContentLanguage.FR -> note
    }

    fun localizedSection(language: ContentLanguage): String? = when (language) {
        ContentLanguage.HE -> sectionHE ?: sectionEN ?: section
        ContentLanguage.EN -> sectionEN ?: section
        ContentLanguage.FR -> section
    }
}
